pint_pandas/pint_array.py

```diff
@@ -288,32 +305,41 @@ class PintArray(ExtensionArray, ExtensionScalarOpsMixin):
     _HANDLED_TYPES = (np.ndarray, numbers.Number, _Quantity)
 
     def __init__(self, values, dtype=None, copy=False):
+        # infer subdtype from values if not given in dtype
+        if (isinstance(dtype, str) and dtype.count("[") <= 1) or isinstance(
+            dtype, _Unit
+        ):
+            _dtype = PintType(dtype)
+            if isinstance(values, _Quantity):
+                values = values.m_as(_dtype.units)
+            values = pd.array(values, copy=copy)
+            dtype = PintType(units=_dtype.units, subdtype=values.dtype)
+        # infer units and subdtype from values if no dtype given
         if dtype is None:
             if isinstance(values, _Quantity):
-                dtype = values.units
+                units = values.units
+                values = pd.array(values, copy=copy)
+                dtype = PintType(units=units, subdtype=values.dtype)
             elif isinstance(values, PintArray):
                 dtype = values._dtype
+
         if dtype is None:
             raise NotImplementedError
 
         if not isinstance(dtype, PintType):
             dtype = PintType(dtype)
         self._dtype = dtype
 
+        # convert units if unit aware values
         if isinstance(values, _Quantity):
             values = values.to(dtype.units).magnitude
         elif isinstance(values, PintArray):
             values = values._data
-        if isinstance(values, np.ndarray):
-            dtype = values.dtype
-            if dtype in ddtypemap:
```

Comment:
As mentioned after the fact with respect to #137, this particular type of dtype handling caused me some grief.  Now that I've sorted out what sort of grief it caused me, I'll be ready to review these changes more properly.